ZuluContent/Engines/Harvest/Mining.cs

```diff
@@ -69,93 +75,170 @@ private Mining()
-                new HarvestResource(80.0, 80.0,  80.0, "You dig some Undead ore and put it in your backpack.", typeof(UndeadOre)),
-                new HarvestResource(85.0, 85.0,  85.0, "You dig some Pyrite ore and put it in your backpack.", typeof(PyriteOre)),
-                new HarvestResource(90.0, 90.0,  90.0, "You dig some Virginity ore and put it in your backpack.", typeof(VirginityOre)),
-                new HarvestResource(95.0, 95.0,  95.0, "You dig some Malachite ore and put it in your backpack.", typeof(MalachiteOre)),
-                new HarvestResource(97.0, 97.0,  97.0, "You dig some Lavarock ore and put it in your backpack.", typeof(LavarockOre)),
-                new HarvestResource(98.0, 98.0,  98.0, "You dig some Azurite ore and put it in your backpack.", typeof(AzuriteOre)),
-                new HarvestResource(100.0, 100.0, 100.0, "You dig some Dripstone ore and put it in your backpack.", typeof(DripstoneOre)),
-                new HarvestResource(104.0, 104.0, 104.0, "You dig some Executor ore and put it in your backpack.", typeof(ExecutorOre)),
-                new HarvestResource(108.0, 108.0, 108.0, "You dig some Peachblue ore and put it in your backpack.", typeof(PeachblueOre)),
-                new HarvestResource(112.0, 112.0, 112.0, "You dig some Destruction ore and put it in your backpack.", typeof(DestructionOre)),
-                new HarvestResource(116.0, 116.0, 116.0, "You dig some Anra ore and put it in your backpack.", typeof(AnraOre)),
-                new HarvestResource(119.0, 119.0, 119.0, "You dig some Crystal ore and put it in your backpack.", typeof(CrystalOre)),
-                new HarvestResource(122.0, 122.0, 122.0, "You dig some Doom ore and put it in your backpack.", typeof(DoomOre)),
-                new HarvestResource(125.0, 125.0, 125.0, "You dig some Goddess ore and put it in your backpack.", typeof(GoddessOre)),
-                new HarvestResource(129.0, 129.0, 129.0, "You dig some New Zulu ore and put it in your backpack.", typeof(NewZuluOre)),
-                new HarvestResource(130.0, 130.0, 130.0, "You dig some Dark Sable Ruby ore and put it in your backpack.", typeof(DarkSableRubyOre)),
-                new HarvestResource(130.0, 130.0, 130.0, "You dig some Ebon Twilight Sapphire ore and put it in your backpack.", typeof(EbonTwilightSapphireOre)),
-                new HarvestResource(140.0, 140.0, 140.0, "You dig some Radiant Nimbus Diamond ore and put it in your backpack.", typeof(RadiantNimbusDiamondOre)),
-            };
-            
+            oreAndStone.Resources = res;
+            oreAndStone.Veins = veins;
 
+            oreAndStone.DefaultVein = new HarvestVein(0.0, new HarvestResource(0.0,
+                "iron",
+                typeof(IronOre)));
 
-            veins = new[]
+            oreAndStone.BonusEffect = (Mobile harvester, Item tool) =>
```

Comment:
You can use a static function as the BonusEffect here, e.g.

```csharp
// Constructor
oreAndStone.BonusEffect = HarvestBonusEffect;
```
```csharp
// Class body
private static void HarvestBonusEffect(Mobile harvester, Item tool)
```

This saves you from defining a massive function in the constructor; also would be a performance issue if this class was being instantiated multiple times.

Reply:
Good point, I'll change it!